CorsixTH/Lua/humanoid_actions/knock_door.lua

```diff
@@ -27,6 +27,11 @@ local KnockDoorAction = _G["KnockDoorAction"]
 --!param door (Object) Door to knock on.
 --!param direction (string) Direction of facing.
 function KnockDoorAction:KnockDoorAction(door, direction)
+  assert(class.is(door, Door), "expecting door object")
+  assert(direction == "north" or direction == "south" or
```

Comment:
I considered it but I don't know where I would put it.

Ideally I have in mind something like class.lua but for simple enums, so we'd write something like ```enum('Direction', { 'north', 'south', 'east', 'west' })``` then everywhere we would write ```enum.is(point.direction, Direction)```

If you don't mind I won't do that for this pr.